test/tests/functional/pbs_test_tpp.py

```diff
@@ -110,38 +227,98 @@ def test_comm_with_mom(self):
-        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
-        self.server.log_match("%s;Exit_status=0" % jid)
-        # Submit multi-chunk job
-        set_attrib = {'Resource_List.select': '2:ncpus=1',
-                      ATTR_l + '.place': 'scatter'}
-        jid = self.submit_job(set_attrib, exp_attrib={'job_state': 'R'},
-                              job_script=True)
-        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
-        self.server.log_match("%s;Exit_status=0" % jid)
-        # Submit interactive job
-        set_attrib = {'Resource_List.select': '2:ncpus=1',
-                      ATTR_inter: '',
-                      ATTR_l + '.place': 'scatter'}
-        jid = self.submit_job(set_attrib, exp_attrib={'job_state': 'R'},
-                              interactive=True)
-        self.server.expect(JOB, 'queue', id=jid, op=UNSET)
-        # Submit reservation
-        set_attrib = {'Resource_List.select': '2:ncpus=1',
-                      ATTR_l + '.place': 'scatter',
-                      'reserve_start': int(time.time() + 10),
-                      'reserve_end': int(time.time() + 120)}
-        exp_attrib = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
-        rid = self.submit_resv(set_attrib, exp_attrib)
-        resv_que = rid.split('.')[0]
-        # Submit job into reservation
-        set_attrib = {'Resource_List.select': '2:ncpus=1',
-                      ATTR_q: resv_que,
-                      ATTR_l + '.place': 'scatter'}
-        jid = self.submit_job(set_attrib, job_script=True)
-        # Wait for reservation to start
-        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
-        self.server.expect(RESV, a, rid, offset=10)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.common_steps(job=True, interactive_job=True, resv=True)
+
+    @requirements(num_moms=2, num_comms=1, num_clients=1)
+    @skipOnCpuSet
+    def test_client_with_mom(self):
+        """
+        This test verifies communication between server-mom,
+        server-client and between moms through pbs_comm
+        Configuration:
+        Node 1 : Server, Mom, Sched, Comm
+        Node 2 : Mom
+        Node 3 : Client
+        """
+        msg = "Need atleast 2 moms and 1 client as input. "
+        msg += "use -pmoms=<m1>:<m2>,client=<c1>"
+        if len(self.moms) < 2 and self.server.client == self.server.hostname:
+            self.skip_test(reason=msg)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        self.hostC = self.server.client
+        nodes = [self.hostA, self.hostB, self.hostC]
+        self.node_list.extend(nodes)
+        self.server.manager(MGR_CMD_SET, SERVER, {'flatuid': True})
+        self.common_steps(job=True, interactive_job=True)
+        self.server.client = self.hostB
```

Comment:
Instead of this, I think we can pass self.hostB as client to common_steps method and then pass it on to submit_job() method's 'client' argument.
or 
If we are altering self.server.client, then at the end of the test, we need to retain the original hostname value.